jaypeasee/overlook-hotel
# Feature: Delete Bookings

# Pull Request Template

#### What’s this PR do?
This PR allows managers to cancel future reservations for guests. Specifically it:
1. Creates `DELETE` Requests when the `[CANCEL]` button is clicked on a future reservation.
1. Tests this method with `Chai Spies`.
1. Removes the `[CANCEL]` button and inserts a cancellation confirmation when a cancellation is made.
1. Renames `apiData` to `apiCalls` for better semantics.

#### Where should the reviewer start?
* Start on the `handleBookingCancellation` function on line 444 of `index.js` to the end of that file.
* Review the `deleteBooking` method on line 35 of `apiCalls.js` to see how the `DELETE` works.
* Review the two tests for `deleteBooking` starting on line 54 of `apiCalls-test.js`

#### How should this be manually tested?
1. Make some bookings and delete them! 
* Watch the `DOM` result of the button removal and new message
* Look at the [API](https://fe-apps.herokuapp.com/api/v1/overlook/1904/bookings/bookings) to confirm they were there before `DELETE`ing and are no longer there after.
* Run `npm test` to make sure all the tests are passing.

#### Any background context you want to provide?
I am successfully `POST`ing and `DELETE`ing but the page isn't updating dynamically. It only updates on refresh. I need to fix this. That will likely be the goal of the next PR.

#### What are the relevant tickets?
* #41 
* #43
